datafusion/src/physical_plan/projection.rs

```diff
@@ -62,18 +61,22 @@ impl ProjectionExec {
     ) -> Result<Self> {
         let input_schema = input.schema();
 
-        let fields: Result<Vec<_>> = expr
+        let fields: Vec<Field> = expr
             .iter()
-            .map(|(e, name)| {
-                Ok(Field::new(
+            .map(|(e, name)| match input_schema.field_with_name(&name) {
+                Ok(f) => f.clone(),
+                Err(_) => Field::new(
                     name,
-                    e.data_type(&input_schema)?,
-                    e.nullable(&input_schema)?,
-                ))
+                    e.data_type(&input_schema).unwrap(),
+                    e.nullable(&input_schema).unwrap_or(true),
```

Comment:
curious what's the reason for these two unwrap changes? For `e.data_type`, I think it would be good to propagate the error back to the call. For `e.nullable(&input_schema)`, I can't think of a reason on why we want to default to true when looking up the corresponding expression metadata from input schema results an error.

Reply:
You’re right left me fix that I don’t think I had any reason